waku/common/databases/db_postgres/dbconn.nim

```diff
@@ -112,23 +122,17 @@ proc waitQueryToFinish(
   ## The 'rowCallback' param is != nil when the underlying query wants to retrieve results (SELECT.)
   ## For other queries, like "INSERT", 'rowCallback' should be nil.
 
-  while db.pqisBusy() == 1:
-    ## TODO: Enhance performance in concurrent queries.
-    ## The connection keeps busy for quite a long time when performing intense concurrect queries.
-    ## For example, a given query can last 11 milliseconds within from the database point of view
-    ## but, on the other hand, the connection remains in "db.pqisBusy() == 1" for 100ms more.
-    ## I think this is because `nwaku` is single-threaded and it has to handle many connections (20)
-    ## simultaneously. Therefore, there is an underlying resource sharing (cpu) that makes this
-    ## to happen. Notice that the _Postgres_ database spawns one process per each connection.
-    let success = db.pqconsumeInput()
-
-    if success != 1:
-      db.check().isOkOr:
-        return err("failed pqconsumeInput: " & $error)
+  var dataAvailable = false
+  proc onDataAvailable(udata: pointer) {.gcsafe, raises: [].} =
+    dataAvailable = true
+
+  let asyncFd = cast[asyncengine.AsyncFD](pqsocket(db))
 
-      return err("failed pqconsumeInput: unknown reason")
+  asyncengine.addReader2(asyncFd, onDataAvailable).isOkOr:
```

Comment:
> AFAIU chronos will trigger `onDataAvailable` directly on a socket read instead of doing `db.check()` which is slower right?

Yes correct :) In fact, the functions `pqGetResult` and `pqConsumeInput` ( [libpq-async](https://www.postgresql.org/docs/current/libpq-async.html) ) doesn't behave well in high load conditions.